### ISSUE TITLE
I shouldn't see warnings in foo.rs for items I allowed in lib.rs (like #[allow(non_snake_case)])

### ISSUE THREAD

**Issue by mrhota** [intellij-rust/intellij-rust#775]
if I have a crate, say, `rustc_llvm`, containing some files like `lib.rs` and `ffi.rs`, Intellij Rust shouldn't complain about non-snake case names in `ffi.rs` if `lib.rs` contains

```
...
#[allow(non_snake_case)]
...
```

Same applies to other `#[allow(...)]` and `#[deny(...)]` uses.

**Comment by mrhota** [comment]
hmmm, I think I had a brain fart about how modules work. Maybe this isn't what users want.

The details rely on faulty premises (`#[allow(...)]` usages in `lib.rs` are somehow related to `foo.rs`), but maybe we can reorient the issue:

Is there some necessary relationship between `#[allow(...)]` usages in a file like `mod.rs` and sibling files like `foo.rs`? I think maybe that's what I was going for originally...

**Comment by alygin** [comment]
I think the right way here would be to follow the same rule `rustc` uses: suppress warnings if an element or one of its parents (transitively, up to the crate root) has the `#[allow(...)]` attribute that is not rewritten by the explicit `#[warn(...)]` attribute at some level.

There already was [a similar issue](https://github.com/intellij-rust/intellij-rust/issues/751) where I promised to implement that for all the naming convention inspections. Unfortunately, I had no time to do that recently, but I'm planning to get back to it a week later.
